### PR TITLE
KIALI-2331 - re-organize cluster roles.

### PR DESCRIPTION
**Pull request by jmazzitelli**
see https://github.com/istio/istio/pull/11234 go in (there is another PR against istio master, too)

This was due to this issue: https://github.com/istio/istio/issues/10883

### CODE REVIEW

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#493](https://paste.fedoraproject.org/paste/T1J3B2ULfLrxC63Pr8Ug1g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1128]](https://paste.fedoraproject.org/paste/6thkWtq7w79OCtgWwm07ZA)

**Comment by chrisob** [comment]
@jmazzitelli Apologies for the confusion! I've created istio/istio#11486 to replicate the same changes to the master branch.

**Comment by jmazzitelli** [comment]
I did a quick smoke test and these changes seem OK.

**Comment by jmazzitelli** [comment]
i think we can merge this. I tested with this and its ok. and it seems that without the "extensions", a crash is caused when deploying on gke. See the istio issue for details.

File: deploy/openshift/clusterrole.yaml [hunk 1]
```diff
@@ -6,22 +6,40 @@ metadata:
...
   - nodes
   - pods
   - projects
   - services
+  - replicationcontrollers
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups: ["extensions", "apps"]
```
Comment (jotak) [inline deploy/openshift/clusterrole.yaml]: maybe "projects" should fall into that apiGroups?

**Comment by chrisob** [comment]
@jotak `kubectl api-resources` is _supremely_ useful when figuring out which resources belong to which groups.

AFAIK, `projects` are an OpenShift-specific resource belonging to the core apiGroup (`""`).

**Comment by jmazzitelli** [comment]
```
$ kubectl api-resources --api-group=project.openshift.io
NAME              SHORTNAMES   APIGROUP               NAMESPACED   KIND
projectrequests                project.openshift.io   false        ProjectRequest
projects                       project.openshift.io   false        Project
```

File: deploy/openshift/clusterrole.yaml (hunk 1 above)
Comment (jotak) [reply deploy/openshift/clusterrole.yaml]: btw do you know how we can get a "topology" of apiGroups & resources registered? This is sometimes confusing in k8s ... objects defined by a CRD are easily trackable, but for others such as "projects" ... I don't know.

**Comment by jotak** [comment]
Ah thanks, didn't see your answers :)

**Comment by jmazzitelli** [comment]
interesting, we had "projects" duplicated :) I'll remove the wrong one in "" group

**Comment by jmazzitelli** [comment]
@jotak we just have `get` for the real projects role - I assume that's all we need (no need for `list` and `watch`?) See: https://github.com/kiali/kiali/blob/master/deploy/openshift/clusterrole.yaml#L96-L100

File: deploy/openshift/clusterrole.yaml (hunk 1 above)
Comment (chrisob) [reply deploy/openshift/clusterrole.yaml]: `kubectl api-resources` lists all API resource types and the apiGroups they belong to.

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#502](https://paste.fedoraproject.org/paste/0A4KA6Xr~Ike39W0awdXUg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1143]](https://paste.fedoraproject.org/paste/rlPQ6lvgf-tWeDj0Z9UGag)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#503](https://paste.fedoraproject.org/paste/2pN0IeMuT-0iMMAXlE1i5Q)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1144]](https://paste.fedoraproject.org/paste/MjWnTxdbYWLdaKrmGxeh~A)

**Comment by jmazzitelli** [comment]
I took off the DNM label - I think this should go in now. These changes look OK and it also fixes the erroneous "projects" role in the "" group. This should also fix the crash that was occurred when deploying as reported upstream.

**Comment by jotak** [comment]
thanks @jmazzitelli

**Pull request merged**